src/cltk/dependency/spacy.py

```diff
@@ -0,0 +1,111 @@
+
+import logging
+import os.path
+from typing import Union, Optional
+
+import spacy
+from spacy.tokens import Doc
+
+from cltk.core.exceptions import (
+    CLTKException,
+    UnimplementedAlgorithmError,
+    UnknownLanguageError,
+)
+from cltk.utils import suppress_stdout
+
+
+LOG = logging.getLogger(__name__)
+LOG.addHandler(logging.NullHandler())
+
+MAP_LANGS_CLTK_SPACY = {
+    "lat": "Latin",
+    # "grc": "Ancient_Greek",
+    # "sa": "Sanskrit"
+}
+
+
+class SpacyWrapper:
+
+    nlps = {}
+
+    def __init__(self,
+                 language: str,
+                 interactive: bool = True,
+                 silent: bool = False):
+        """
+
+        :param language:
+        :param interactive:
+        :param silent:
+        """
+        self.language = language
+        self._interactive = interactive
+        self._silent = silent
+
+        if self._interactive and self._silent:
+            raise ValueError(
+                "``interactive`` and ``silent`` options are not compatible with each other."
+            )
+
+        if not self.is_wrapper_available:
+            raise UnknownLanguageError(f"Language {self.language} is not supported by CLTK wrapper of SpaCy.")
+
+        if not self._is_model_present:
+            self._download_model()
+
+        with suppress_stdout():
+            self.nlp = self._load_pipeline()
+            self.nlps[self.language] = self.nlp
+
+    @property
+    def is_wrapper_available(self) -> bool:
+        return self.language in MAP_LANGS_CLTK_SPACY
+
+    def _load_pipeline(self) -> Optional[spacy.Language]:
```

Comment:
Typing this as `Optional` is misleading as the function either returns a `Language` or throws an exception but never returns with `None`

Reply:
True.